multilep/src/GenAnalyzer.cc

```diff
@@ -105,8 +125,84 @@ void GenAnalyzer::analyze(const edm::Event& iEvent){
                 _gen_phPassParentage[_gen_nPh] = GenTools::passParentage(p, *genParticles);
                 ++_gen_nPh;
             } 
-        }
-    }
+	}
+
+       //store all generator level particles
+       if( multilepAnalyzer->storeGenParticles )
+	 {	    
+	    float ptGen = p.pt();
+	    float etaGen = p.eta();
+	    float phiGen = p.phi();
+	    float EGen = p.energy();
+	    int idGen = p.pdgId();
+	    int statusGen = p.status();
+	    int chargeGen = p.charge();
+	    bool isPromptFinalStateGen = p.isPromptFinalState();
+	    bool isDirectPromptTauDecayProductFinalState = p.isDirectPromptTauDecayProductFinalState();
+	    bool isLastCopy = p.isLastCopy();
+	    int indexGen = _gen_n;
+
+	    if( _gen_n == gen_n_max )
+	      {		      
+		 std::cout << "Reached max" << std::endl;
+		 exit(1);
+	      }
```

Comment:
Maybe you could consider implementing an exception in this style: https://github.com/GhentAnalysis/heavyNeutrino/blob/master/multilep/src/LheAnalyzer.cc#L75-L77

Reply:
One can just do "throw std::out_of_range( "Reached max" )"